### PR TITLE
Add convenience anonymousClassBuilder(CodeBlock)

### LINKED ISSUE #558: Proposal: Add a convenience `anonymousClassBuilder` method to TypeSpec that accepts a CodeBlock
In some cases - I already have a formatted `CodeBlock` (either borrowed from another `TypeSpec` with the same signature or otherwise). If this sounds good, I can make a PR

### PR DESCRIPTION
**Pull request by hzsweers**
Resolves #558

### CODE REVIEW

File: src/main/java/com/squareup/javapoet/TypeSpec.java [hunk 1]
```diff
@@ -111,9 +111,13 @@ public static Builder enumBuilder(ClassName className) {
   }
 
   public static Builder anonymousClassBuilder(String typeArgumentsFormat, Object... args) {
-    return new Builder(Kind.CLASS, null, CodeBlock.builder()
-        .add(typeArgumentsFormat, args)
-        .build());
+    return anonymousClassBuilder(CodeBlock.builder()
+            .add(typeArgumentsFormat, args)
+            .build());
```
Comment (JakeWharton) [inline src/main/java/com/squareup/javapoet/TypeSpec.java]: Wrong indent

Comment (hzsweers) [inline src/main/java/com/squareup/javapoet/TypeSpec.java]: derp, will fix

Comment (hzsweers) [inline src/main/java/com/squareup/javapoet/TypeSpec.java]: 9da74b5

**Comment by hzsweers** [comment]
Thanks!

**Pull request merged**